lib/api/src/sys/instance.rs

```diff
@@ -128,6 +128,14 @@ impl Instance {
                 (name, extern_)
             })
             .collect::<Exports>();
+        
+        // If the memory is imported then also export it for backwards compatibility reasons
+        // (many will assume the memory is always exported) - later we can remove this
+        if exports.get_memory("memory").is_err() {
+            if let Some(memory) = externs.iter().filter(|a| a.ty(store).memory().is_some()).next() {
+                exports.insert("memory", memory.clone());
+            }
+        }
 
```

Comment:
does this change is also a break-change on the API level?

Reply:
this probably needs some more thought on it anyway as it assumes the first exported memory but in the future there could be multiple - it does not change the API but does change the behavior.